### ISSUE TITLE
Issues with method references in teavm

### ISSUE THREAD

**Issue by Ihromant** [konsoletyper/teavm#861]
I found this issue on very simple case, but I suppose that something in method references support in teavm is broken.
Consider following test:
```java
@Test
    public void testComparator() {
        Map<Integer, Double> distances = new HashMap<>();
        distances.put(1, 5.0);
        distances.put(2, 2.0);
        distances.put(3, 3.0);
        distances.put(4, 4.0);
        assertEquals(Integer.valueOf(2), IntStream.rangeClosed(1, 4).boxed().min((i, j) -> Double.compare(distances.get(i), distances.get(j))).orElseThrow());
        assertEquals(Integer.valueOf(2), IntStream.rangeClosed(1, 4).boxed().min(Comparator.comparingDouble(distances::get)).orElseThrow());
    }
```
It fails in the second line despite implementation in TComparator is same as in unwrapped in first line. Firstly I thought that there is minor misprint in classlib, but I suppose that it's more complex issue that should be fixed globally.

**Comment by konsoletyper** [comment]
Did you try to debug it?

**Comment by Ihromant** [comment]
No, I assumed that it's somewhere deep after reviewing TComparator implementation (the onlyp lace where issue could be in classlib). Will take a look on generated JS, maybe there is a hint.

**Comment by konsoletyper** [comment]
But what about placing a breakpoint on an arbitrary line, stepping, etc?

**Comment by Ihromant** [comment]
In JS or where?

**Comment by konsoletyper** [comment]
On JS, for example

**Comment by Ihromant** [comment]
I'm usually trying to avoid debugging JS. But ok, will try to debug this issue.

**Comment by Ihromant** [comment]
I was able to locate what causes the issue.
As I suspected, it is something wrong with method references.
```java
List<Double> doubles = List.of(1.0, 2.0, 3.0);
ToDoubleFunction<Integer> f = doubles::get;
assertEquals(Double.valueOf(2.0), (Double) f.applyAsDouble(1)); // java.lang.RuntimeException: java.lang.AssertionError: expected:<2.0> but was:<NaN>
List<Integer> integers = List.of(1, 2, 3);
ToIntFunction<Integer> f1 = integers::get;
assertEquals(Integer.valueOf(2), (Integer) f1.applyAsInt(1)); // java.lang.RuntimeException: java.lang.AssertionError: expected:<2> but was:< > (here is some strange symbol)
```

**Comment by Ihromant** [comment]
Also I think that this fix should be merged with 0.9.0 and released as 0.9.1.
